packages/x-date-pickers/src/tests/describeGregorianAdapter/testCalculations.ts

```diff
@@ -164,122 +277,343 @@ export const testCalculations: DescribeGregorianAdapterTestSuite = ({ adapter })
     });
   });
 
-  it('Method: isEqual', () => {
-    expect(adapter.isEqual(adapter.date(null), null)).to.equal(true);
-    expect(adapter.isEqual(testDateIso, adapter.date(TEST_DATE_ISO_STRING))).to.equal(true);
-    expect(adapter.isEqual(null, testDateIso)).to.equal(false);
-    expect(adapter.isEqual(testDateLocale, adapter.date(TEST_DATE_LOCALE_STRING))).to.equal(true);
-    expect(adapter.isEqual(null, testDateLocale)).to.equal(false);
+  describe('Method: isEqual', () => {
+    it('should work in the same timezone', () => {
+      expect(adapter.isEqual(adapter.date(null), null)).to.equal(true);
+      expect(adapter.isEqual(testDateIso, adapter.date(TEST_DATE_ISO_STRING))).to.equal(true);
+      expect(adapter.isEqual(null, testDateIso)).to.equal(false);
+      expect(adapter.isEqual(testDateLocale, adapter.date(TEST_DATE_LOCALE_STRING))).to.equal(true);
+      expect(adapter.isEqual(null, testDateLocale)).to.equal(false);
+    });
+
+    it('should work with different timezones', function test() {
+      if (!adapter.isTimezoneCompatible) {
+        this.skip();
+      }
+
+      const londonTimezone = adapterTZ.setTimezone(testDateIso, 'Europe/London');
+      const parisTimezone = adapterTZ.setTimezone(testDateIso, 'Europe/Paris');
+
+      expect(adapterTZ.isEqual(londonTimezone, parisTimezone)).to.equal(true);
+    });
   });
 
-  it('Method: isSameYear', () => {
-    expect(adapter.isSameYear(testDateIso, adapter.date('2018-10-01T00:00:00.000Z')!)).to.equal(
-      true,
-    );
-    expect(adapter.isSameYear(testDateIso, adapter.date('2019-10-01T00:00:00.000Z')!)).to.equal(
-      false,
-    );
-    expect(adapter.isSameYear(testDateLocale, adapter.date('2018-10-01T00:00:00.000Z')!)).to.equal(
-      true,
-    );
-    expect(adapter.isSameYear(testDateLocale, adapter.date('2019-10-01T00:00:00.000Z')!)).to.equal(
-      false,
-    );
+  describe('Method: isSameYear', () => {
+    it('should work in the same timezone', () => {
+      expect(adapter.isSameYear(testDateIso, adapter.date('2018-10-01T00:00:00.000Z')!)).to.equal(
+        true,
+      );
+      expect(adapter.isSameYear(testDateIso, adapter.date('2019-10-01T00:00:00.000Z')!)).to.equal(
+        false,
+      );
+      expect(
+        adapter.isSameYear(testDateLocale, adapter.date('2018-10-01T00:00:00.000Z')!),
+      ).to.equal(true);
+      expect(
+        adapter.isSameYear(testDateLocale, adapter.date('2019-10-01T00:00:00.000Z')!),
+      ).to.equal(false);
+    });
+
+    it('should work with different timezones', function test() {
+      if (!adapter.isTimezoneCompatible) {
+        this.skip();
+      }
+
+      // Both dates below have the same timestamp, but they are not in the same year when represented in their respective timezone.
+      // The adapter should still consider that they are in the same year.
```

Comment:
This comment is somewhat disturbing. Would it be true to say that the `isSameYear` relies on the timezone of the first/second date provided?

Reply:
>  Would it be true to say that the isSameYear relies on the timezone of the first/second date provided?

Yes, all the `isSameXXX` `isAfterXXX` and `isBeforeXXX` are applied on the timezone of the 1st param.
`isSameYear` answers the question "for any A and B, is A and B in the same year when displayed in the timezone of A"